### DISCUSSION TITLE
Escalation to TheHive failed

### DISCUSSION THREAD

**Discussion by abs0248** [Security-Onion-Solutions/securityonion#2337] [Unsupported Versions] [answered]
I tried the Alert Escalation to TheHive without any problem for a week.now when to try to escalate from SOC , i got the following error on top of SOC  :
`Request did not complete successfully (401): 401 Unauthorized`
also the SOC log file show as below : 
`level=error message="Failed request" contentLength=0 elapsedMs=31 error="Request did not complete successfully (401): 401 Unauthorized" impl=*server.CaseHandler method=POST path=/api/case sourceIp=172.17.0.1:54814 statusCode=400`
I can also successfully login to TheHive and can create a case manually without any problem.Any idea what is broken?

**Comment by jertel** [comment, marked as answer]
Did you (or someone else) revoke or renew the API key inside TheHive?

**Reply by abs0248** [reply]
No i never touch that.I think the only change i made that cloud break it , was PROXY setting i made on the ENV and Dockers pillar to make the `so-rule-update` work behind the proxy.i changed the proxy setting with var like `no_proxy=172.16.0.0/16,172.17.0.0/16` so that if any API communicating each others over docker0 bridge network , bypassing the proxy setting.No success with those setting.i need to put some time for T-shoot the issue.

**Reply by weslambert** [reply]
You may want to manually try hitting the TheHive API from the box itself, using something like:

```
curl -XPOST -H 'Authorization: Bearer ***API*KEY***' -H 'Content-Type: application/json' http://localhost:9000/api/case -d '{
  "title": "My first case",
  "description": "This case has been created by my custom script"
}'
```

**Reply by abs0248** [reply]
Thank you.your advice let me find the problem.i tried the API call you suggested with no success . i tried to query for a case that i manually created  using the  Thehive API and again with no success.the problem was after a week of SO installation i created a new User account as my primary SO account  and disabled the previous user account both on the SO and Thehive UI.the API-KEY on the global.sls pointed to the previous and now disabled user and the newly created user didn't have any  API-KEY enabled/created on the UI of TheHive (UI/Users) . So on Thehive UI i generate a new API-KEY for this newly created user and copy/paste the KEY to the global.sls and now it is working.it look like that creating a new user and disabling the old one didn't update the API-KEY on the salt global philar.i think the case can be closed.

**Reply by ograso** [reply]
Hey abs0248,
Can you explain how to create API-KEY for newly created users?? And replace with the "cortexuserkey: xxx" in global.sls. Right?
